lib/ex_unit/test/ex_unit/doc_test_test.exs

```diff
@@ -476,14 +545,50 @@ defmodule ExUnit.DocTestTest do
         doctest ExUnit.DocTestTest.IndentationNotEnough
       end
     end
+  end
 
+  test "fails in fenced indentation mismatch" do
+    assert_raise ExUnit.DocTest.Error,
+      ~r[test/ex_unit/doc_test_test\.exs:\d+: indentation level mismatch: " iex> bar = 2", should have been 0 spaces], fn ->
+      defmodule NeverCompiled do
+        import ExUnit.DocTest
+        doctest ExUnit.DocTestTest.FencedIndentationMismatchedPrompt
+      end
+    end
+
+    assert_raise ExUnit.DocTest.Error,
+      ~r[test/ex_unit/doc_test_test\.exs:\d+: indentation level mismatch: "  3", should have been 0 spaces], fn ->
+      defmodule NeverCompiled do
+        import ExUnit.DocTest
+        doctest ExUnit.DocTestTest.FencedIndentationTooMuch
+      end
+    end
+
+    assert_raise ExUnit.DocTest.Error,
+      ~r[test/ex_unit/doc_test_test\.exs:\d+: indentation level mismatch: \"3\", should have been 2 spaces], fn ->
+      defmodule NeverCompiled do
+        import ExUnit.DocTest
+        doctest ExUnit.DocTestTest.FencedIndentationNotEnough
+      end
+    end
+  end
+
+  test "fails with improper termination" do
     assert_raise ExUnit.DocTest.Error,
       ~r[test/ex_unit/doc_test_test.exs:\d+: expected non-blank line to follow iex> prompt], fn ->
```

Comment:
hi Chris: do not worry about the merging conflicts, I will take care of them and I will wait for your PR to be merged. 
Sorry about this last comment, it was not the code you introduced, so just forget about it.

Regarding doing rebase, it is perfectly fine (I do that all the time). You can also add it as a commit. Github now allows to squash commit into one, when merging. it's up you I guess, as I have seen people using both here.